src/datachain/data_storage/warehouse.py

```diff
@@ -200,23 +200,17 @@ def dataset_row_cls(self) -> type["DataTable"]:
     def dataset_select_paginated(
         self,
         query,
-        limit: Optional[int] = None,
-        order_by: tuple["ColumnElement[Any]", ...] = (),
         page_size: int = SELECT_BATCH_SIZE,
-    ) -> Generator[RowDict, None, None]:
+    ) -> Generator[Sequence, None, None]:
         """
         This is equivalent to `db.execute`, but for selecting rows in batches
         """
-        cols = query.selected_columns
-        cols_names = [c.name for c in cols]
+        limit = query._limit
+        paginated_query = query.limit(page_size)
 
-        if not order_by:
-            ordering = [cols.sys__id]
-        else:
-            ordering = order_by  # type: ignore[assignment]
-
-        # reset query order by and apply new order by id
-        paginated_query = query.order_by(None).order_by(*ordering).limit(page_size)
+        if not paginated_query._order_by_clauses:
```

Comment:
is there any improvement in this or it's a more or less cosmetic?

Reply:
It is more cosmetic and refactoring to simplify things.

Reply:
sounds good, but please consider doing this things in a separate PR (can be a chained PR) ... or split by commits ... it makes it harder to review otherwise 🙏

Reply:
I know, I am trying to do my best but it is hard to resist sometimes :) Will isolate some changes to separate PR 👍